dbms/src/Databases/DatabasesCommon.cpp

```diff
@@ -184,22 +186,19 @@ void DatabaseWithOwnTablesBase::attachTable(const String & table_name, const Sto
 }
 
 
-void DatabaseWithOwnTablesBase::attachDictionary(const String & dictionary_name, const Context & context, bool load)
+void DatabaseWithOwnTablesBase::attachDictionary(const String & dictionary_name, const Context & context)
 {
-    const auto & external_loader = context.getExternalDictionariesLoader();
-
     String full_name = getDatabaseName() + "." + dictionary_name;
     {
         std::lock_guard lock(mutex);
-        auto status = external_loader.getCurrentStatus(full_name);
-        if (status != ExternalLoader::Status::NOT_EXIST || !dictionaries.emplace(dictionary_name).second)
-            throw Exception(
-                      "Dictionary " + full_name + " already exists.",
-                      ErrorCodes::DICTIONARY_ALREADY_EXISTS);
+        if (!dictionaries.emplace(dictionary_name).second)
+            throw Exception("Dictionary " + full_name + " already exists.", ErrorCodes::DICTIONARY_ALREADY_EXISTS);
```

Comment:
Why we removed this check?

Reply:
`createDictionary()` adds a temp metadata file with a dictionary to `ExternalLoader`, then loads it (if `lazy_load == false`), then calls `attachDictionary()`. At the moment when `attachDictionary()` is called the dictionary could be already loaded. I don't think this check was very useful before anyway.